channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -670,29 +686,64 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                 try {
                     let onCompleteFuncs = {
                         xdrOnComplete: (xdr: IXDomainRequest, oncomplete: OnCompleteCallback,payload?: IPayloadData) => {
-                            let data = _getPayloadArr(payload);
-                            if (!data) {
+                            let payloadArr = _getPayloadArr(payload);
+                            if (!payloadArr) {
                                 return;
                             }
-                            return _xdrOnLoad(xdr, data);
+                            const responseText = _getResponseText(xdr);
+                            let statsbeat = _getStatsBeat();
+                            if (statsbeat) {
+                                var endpointHost = urlParseUrl(_self._senderConfig.endpointUrl).hostname;
```

Comment:
This should probably be using the `_endpointUrl` so that when redirects are detected and updated the correct URL that is being used is used.  See the `_checkAndUpdateEndPointUrl` function which is where it "updates" this internal endpoint URL and how it detects that breeze sent us a redirect.